x-govuk/components/secondary-navigation/_secondary-navigation.scss

```diff
@@ -0,0 +1,102 @@
+.x-govuk-secondary-navigation {
+  @include govuk-font(19);
+
+  @include govuk-media-query($from: tablet) {
+    display: flex;
+    overflow-x: auto;
+    padding-top: govuk-spacing(2);
+    padding-bottom: govuk-spacing(7);
+  }
+
+  @include govuk-media-query($until: tablet) {
+    display: block;
+  }
+}
+
+.x-govuk-secondary-navigation__list {
+  width: 100%;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  @include govuk-media-query($from: tablet) {
+    display: flex;
+    margin: 0;
+    white-space: nowrap;
+
+    box-shadow: inset 0 -1px 0 $govuk-border-colour;
+  }
+
+  @include govuk-media-query($until: tablet) {
+    margin-bottom: govuk-spacing(4);
+  }
+}
+
+.x-govuk-secondary-navigation__list-item {
+  @include govuk-media-query($from: tablet) {
+    display: flex;
+
+    & + & {
+      margin-left: govuk-spacing(4);
+    }
+  }
+
+  a,
+  a:link,
+  a:visited {
+    background-color: inherit;
+    display: block;
+    text-decoration: none;
+
+    @include govuk-media-query($from: tablet) {
+      border-bottom: 5px solid transparent;
+      padding-bottom: govuk-spacing(3) - 5px; // Compensate for 5px border
+    }
+
+    @include govuk-media-query($until: tablet) {
+      background-color: inherit;
+      border-left: 5px solid transparent;
+      padding: govuk-spacing(2);
+    }
+  }
+
+  a:focus {
+    color: $govuk-focus-text-colour;
+    background-color: $govuk-focus-colour;
+    box-shadow: 0 -2px $govuk-focus-colour, 0 5px $govuk-focus-text-colour;
+    border-color: $govuk-focus-text-colour;
+    border-left-color: transparent;
+    position: relative;
+  }
+}
+
+.x-govuk-secondary-navigation__list-item--current {
+  a,
+  a:link,
+  a:visited {
+    border-color: $govuk-brand-colour;
+    color: $govuk-text-colour;
+  }
+
+  a:focus {
+    color: $govuk-focus-text-colour;
+    background-color: $govuk-focus-colour;
+    box-shadow: 0 -2px $govuk-focus-colour, 0 5px $govuk-focus-text-colour;
+    border-color: $govuk-focus-text-colour;
+    border-left-color: transparent;
```

Comment:
I’d be tempted to keep this using the link focus style here, and for a number of reasons:

1. In GOV.UK Frontend, there’s discussion around having many different focus styles (weighed up against ensuring each focus style is distinct enough from the unfocused state for a given component). This is yet another focus style.

2. It’s visually inconsistent with the primary navigation focus style:
  
    <img width="300" alt="Screenshot of focus state in primary navigation" src="https://github.com/x-govuk/govuk-prototype-components/assets/813383/9aed7cd8-7017-4421-abfc-c3413da56538">

3. It adds additional styles where we can lean on either existing CSS via the cascade or using existing link `@mixins`

Reply:
I'm not sure about the focus and hover states either. For the primary nav, the trend seems to be moving towards the focus area being bigger, making it the same size as the clickable area, rather than smaller.

Eg the [Design System website](https://design-system.service.gov.uk) uses the smaller focus area, but the [One Login header](https://govuk-one-login.github.io/service-header/dist/preview.html) uses the larger one, as does the [MOJ primary navigation](https://design-patterns.service.justice.gov.uk/components/primary-navigation/).

Personally, I think I prefer the larger focus area, mainly to avoid the double-border below the text? 🤷 

Agree that the primary and secondary nav should be consistent though...

Reply:
The problem with the bigger focus area is that it’s not distinct from a currently selected item. For example, removing all colour, a focused item looks like a currently highlighted item:

* GOV.UK Design System navigation (current versus focused):

    ![Screenshot of navigation from GOV.UK Design System website.](https://github.com/x-govuk/govuk-prototype-components/assets/813383/53ad5532-c648-488f-a831-12e09784e5a5)

* GOV.UK One Login service header example (focused versus current):

    ![Screenshot of navigation from GOV.UK One Login service header.](https://github.com/x-govuk/govuk-prototype-components/assets/813383/228d5abf-33cb-4b1c-a89a-26da0229e52f)

I note that you increased the bottom border on a focused area which avoid this issue, but it introduces a whole new focus style. Perhaps something to file an issue on upstream on the One login repo?

For now, and for the sake of simplicity and following current convention, perhaps we use the link focus styles in the secondary navigation?

Reply:
I've done a big refactor in 26adff173b14d8c85f3ab1e7faaa58c8e8b66b91 which reverts the css to be more similar to the primary navigation, with the focus highlight only extending to the text, and the underline on hover also being just below the text rather than where the selected item border is.

I do still think that having the double-border is a bit visually-jarring, but I agree we should keep the secondary nav consistent with the primary one, and then if there's any changes we could apply them to both.